bayes_opt/acquisition.py

```diff
@@ -1,18 +1,25 @@
 """Acquisition functions for Bayesian Optimization."""
-import warnings
+
+from __future__ import annotations
+
 import abc
+import warnings
+from copy import deepcopy
+from numbers import Number
+from typing import TYPE_CHECKING, Callable
+
 import numpy as np
 from numpy.random import RandomState
 from scipy.optimize import minimize
-from scipy.stats import norm
 from scipy.special import softmax
-from .target_space import TargetSpace
-from .constraint import ConstraintModel
+from scipy.stats import norm
 from sklearn.gaussian_process import GaussianProcessRegressor
-from typing import Callable, List, Union, Tuple
-from copy import deepcopy
-from numbers import Number
-from .util import ConstraintNotSupportedError, NoValidPointRegisteredError, TargetSpaceEmptyError
+
+from bayes_opt.target_space import TargetSpace
+from bayes_opt.util import ConstraintNotSupportedError, NoValidPointRegisteredError, TargetSpaceEmptyError
+
+if TYPE_CHECKING:
+    from bayes_opt.constraint import ConstraintModel
```

Comment:
Why is this condition necessary?

Reply:
This is because it is not used during runtime.
By importing it under the `TYPE_CHECKING` condition,
we can avoid circular reference errors that might occur in the future.

I haven't checked, but it is probably not necessary.